src/CommunityToolkit.Maui.SourceGenerators/Extensions/CompilationExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace CommunityToolkit.Maui.SourceGenerators.Extensions
+{
+	static class CompilationExtensions
+	{
+		public static TSymbol? GetSymbol<TSymbol>(this Compilation compilation, BaseTypeDeclarationSyntax declarationSyntax)
+			where TSymbol : ISymbol
+		{
+			var model = compilation.GetSemanticModel(declarationSyntax.SyntaxTree);
+			return (TSymbol?)model.GetDeclaredSymbol(declarationSyntax);
```

Comment:
Can this cast fail in error scenarios?

Reply:
What will be the error scenario for this context?

Reply:
@pictos No specific scenario in mind, but I think this can happen. Using `as TSymbol` instead of a direct cast is much safer.

Reply:
I would say that we want this to fail. If the type isn't the wanted we should see an exception